src/test/java/com/github/lipinskipawel/board/ai/bruteforce/MiniMaxTest.java

```diff
@@ -65,5 +63,23 @@ void notScoreAGoal() {
                     () -> Assertions.assertThat(afterAiMove.getPlayer()).isEqualByComparingTo(Player.FIRST)
             );
         }
+
+        @Test
+        @DisplayName("SmartBoardEvaluator should beat DummyBoardEvaluator")
+        void checkBetterEvaluator() {
+            final var dummyEvaluator = new DummyBoardEvaluator();
+            final var smartEvaluator = new SmartBoardEvaluator();
+
+            var gameBoard = board;
+            BoardEvaluator evaluator = dummyEvaluator;
```

Comment:
can it be `var`?

Reply:
No, because it's gonna evaluate it as DummyBoardEvaluator not BoardEvaluator. It's making problem in line 79.